libnetdata/socket/security.c

```diff
@@ -66,7 +66,7 @@ int tls_select_version(const char *lversion) {
         return TLS1_1_VERSION;
     else if (!strcmp(lversion, "1.2"))
         return TLS1_2_VERSION;
-#if (OPENSSL_VERSION_NUMBER >= OPENSSL_VERSION_111 && !defined(LIBRESSL_VERSION_NUMBER)) || ((defined(LIBRESSL_HAS_TLS1_3) || defined(LIBRESSL_INTERNAL)))
+#if (OPENSSL_VERSION_NUMBER >= OPENSSL_VERSION_111 && !defined(LIBRESSL_VERSION_NUMBER)) || (defined(LIBRESSL_HAS_TLS1_3))
```

Comment:
Note that also the OPENSSL_* check should check **features** (TLS1_3_VERSION and related) not second-guess from versions. I don't have a vote on the upstream code, but if I did, I'd veto this and request revision.